### ISSUE TITLE
Feature request: Include a timestamp  in the return struct of the `fetch_installation_keys` RPC

### ISSUE THREAD

**Issue by insipx** [xmtp/xps-gateway#61]
### Is your feature request related to a problem?

libxmtp filters installation keys by the timestamp they were committed

### Describe the solution to the problem

We should return the block time in nanoseconds of the event in which the `get_fetch_keys` operation occurred

### Describe the uses cases for the feature

allow libxmtp to get identity updates only after a certain time

**Comment by jac18281828** [comment]
Does it make sense to consider ‘seconds’ for this?   Block times are seconds since epoch.   These times can also differ from real time by a large delta.   ‘Nanoseconds’ may lead to a belief that these times are more meaningful than they are.

**Comment by insipx** [comment]
Yeah I don't think it's a great metric by any means, but it's consistent with how libxmtp filters through `installation_ids`. I'll leave it in ns for the demo but it would make sense not to use timestamp at all in libxmtp and instead use block height and log index. 

I think that seconds could be used instead of nanoseconds, libxmtp doesn't check for it, so I'll experiment with that too
